skfda/preprocessing/dim_reduction/projection/_fpca.py

```diff
@@ -1,33 +1,445 @@
+        """Computes the n_components first principal components and their scores
+        and returns them.
+        Args:
+            X (FDataGrid or FDataBasis):
+                the functional data object to be analysed
+            y (None, not used):
+                only present for convention of a fit function
+
+        Returns:
+            (array_like): the scores of the data with reference to the
+            principal components
+        """
+        self.fit(X, y)
+        return self.transform(X, y)
+
+
+class FPCABasis(FPCA):
+    """Functional principal component analysis for functional data represented
+    in basis form.
+
+    Attributes:
+        components_ (FDataBasis): this contains the principal components in a
+            basis representation.
+        component_values_ (array_like): this contains the values (eigenvalues)
+            associated with the principal components.
+        pca_ (sklearn.decomposition.PCA): object for PCA.
+            In both cases (discretized FPCA and basis FPCA) the problem can be
+            reduced to a regular PCA problem and use the framework provided by
+            sklearn to continue.
+
+    Parameters:
+        n_components (int): number of principal components to obtain from
+            functional principal component analysis. Defaults to 3.
+        centering (bool): if True then calculate the mean of the functional data
+            object and center the data first. Defaults to True. If True the
+            passed FDataBasis object is modified.
+        components_basis (Basis): the basis in which we want the principal
+            components. We can use a different basis than the basis contained in
+            the passed FDataBasis object.
+        regularization_lfd (LinearDifferentialOperator, list or int): Linear
+                differential operator. If it is not a LinearDifferentialOperator
+                object, it will be converted to one. If you input an integer
+                then the derivative of that degree will be used to regularize
+                the principal components.
+
+    Examples:
+        Construct an artificial FDataBasis object and run FPCA with this object.
+        The resulting principal components are not compared because there are
+        several equivalent possibilities.
+
+        >>> data_matrix = np.array([[1.0, 0.0], [0.0, 2.0]])
+        >>> sample_points = [0, 1]
+        >>> fd = FDataGrid(data_matrix, sample_points)
+        >>> basis = skfda.representation.basis.Monomial((0,1), n_basis=2)
+        >>> basis_fd = fd.to_basis(basis)
+        >>> fpca_basis = FPCABasis(2)
+        >>> fpca_basis = fpca_basis.fit(basis_fd)
+
+    """
+
+    def __init__(self,
+                 n_components=3,
+                 components_basis=None,
+                 centering=True,
+                 regularization_parameter=0,
+                 regularization_lfd=2):
+        """FPCABasis constructor
```

Comment:
I think that it is better not to repeat the parameters in the docstring of the `__init__` method, as scikit-learn does.

Reply:
Done